magefiles/rulesengine/repos/infra_deployments.go

```diff
@@ -1,17 +1,170 @@
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+
+		rctx.LabelFilter = fmt.Sprintf("%s,%s", rctx.LabelFilter, "image-controller")
+
+		return nil
+
+	})}}
+
+var InfraDeploymentsMultiPlatformTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Multi Controller component File Change Rule",
+	Description: "Map multi platform tests files when Multi Controller component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/multi-platform-controller/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+
+		rctx.LabelFilter = fmt.Sprintf("%s,%s", rctx.LabelFilter, "multi-platform")
+
+		return nil
+
+	})}}
+
+var InfraDeploymentsBuildTemplatesTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Build-templates component File Change Rule",
+	Description: "Map build-templates tests files when Build-templates component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/build-templates/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+
+		rctx.LabelFilter = fmt.Sprintf("%s,%s", rctx.LabelFilter, "build-templates")
+
+		return nil
+
+	})}}
+
+var InfraDeploymentsBuildServiceTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Build service component File Change Rule",
+	Description: "Map build service tests files when Build service component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/build-service/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+
+		rctx.LabelFilter = fmt.Sprintf("%s,%s", rctx.LabelFilter, "build-service")
+
+		return nil
+
+	})}}
+
+var InfraDeploymentsReleaseServiceTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Release service component File Change Rule",
+	Description: "Map release service tests files when Release service component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/release/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+
+		rctx.LabelFilter = fmt.Sprintf("%s,%s", rctx.LabelFilter, "release-service")
+
+		return nil
+
+	})}}
+
+// CheckNoComponentsFilesChanged checks if in repo infra-deployments changed files other than components
+func CheckNoComponentsFilesChanged(rctx *rulesengine.RuleCtx) bool {
```

Comment:
There are two different patterns being used for the default rule and the diff files rule. Neither is right or wrong. This comment is more of a consistency of following the same pattern for both rules set. So this one is more of a nitpick. Since you've already written out the rules for each directory, you may want to convert the default rule into a RuleChain as well and us a `None` conditional check around all the rules.

Reply:
good point  , 
if I went to implement the default rule based on the RuleChain you mentioned,  
would it it be like the suggested below ? 

```
var InfraDeploymetsDefaultRuleChain = rulesengine.Rule{Name: "Infra Deployments Default Test Execution",
	Description: "Run the Konflux-demo suite tests when an Infra-deployments PR includes changes to files outside of the specified components.",
	Condition:    rulesengine.None{&InfraDeploymentsTestFilesOnlyRule},
	Actions:     []rulesengine.Action{rulesengine.ActionFunc(ExecuteInfraDeploymentsDefaultTestAction)},
}
```

Reply:
@kasemAlem I was thinking something like the below. That is what I have in the example from the e2e-test repo

```
                rulesengine.None{
                        rulesengine.ConditionFunc(CheckPkgFilesChanged),
                        rulesengine.ConditionFunc(CheckMageFilesChanged),
                        rulesengine.ConditionFunc(CheckCmdFilesChanged),
                        rulesengine.ConditionFunc(CheckNoFilesChanged),
                        rulesengine.ConditionFunc(CheckTektonFilesChanged),
                },
```

Reply:
got it, 
I've replaced it to use the current rules with **None** , 
also tested